patches/minecraft/net/minecraft/inventory/SlotFurnaceOutput.java.patch

```diff
@@ -1,11 +1,18 @@
 --- ../src-base/minecraft/net/minecraft/inventory/SlotFurnaceOutput.java
 +++ ../src-work/minecraft/net/minecraft/inventory/SlotFurnaceOutput.java
-@@ -83,6 +83,8 @@
+@@ -37,6 +37,7 @@
  
+     public void func_82870_a(EntityPlayer p_82870_1_, ItemStack p_82870_2_)
+     {
++        if(field_75228_b>0)
+         this.func_75208_c(p_82870_2_);
```

Comment:
Actually, the more I look into this. The more that the call to onCrafting in onPickupFromSlot is not needed at all. the onPickupFromSlot is ACTUALLY something more like onSlotChanged. With the ItemStack being the remaining itemstack. We don't want to send that out.
All cases in ContainerFurnace.transferStackInSlot where you're transferring OUT of slot 2 {The output} are covered. What we need to do is remove the onCrafting call from onPickupFromSlot  entirely and find the part that covers clicking instead of shift-clicking and see if that can be made to work properly if it doesn't already.